Add VSCode extension documentation (#20385)
MystenLabs/sui: docs/content/references/ide/move.mdx

```bash
code --install-extension mysten.move
```

### Installing `move-analyzer`

The Move extension will attempt to install the appropriate `move-analyzer` binary for your platform. If this doesn't work, or you would prefer to install it manually, you can build it with Cargo:

```bash
cargo install --git https://github.com/MystenLabs/sui.git sui-move-lsp
```
By default, the Move extension expects to find the `move-analyzer` binary in `~/.sui/bin`. You can either copy the binary here, or configure the extension to use a different path. 

## Features

### Build, Test, and Trace

The Move extension installs command palette commands for building, testing, and tracing Move code. 

![Move commands in the command palette](./images/commands.png)

These commands will attempt to find the `Move.toml` file for the open Move source file and will open a terminal to run the appropriate `sui move` command.

In order to generate a trace, you will need to have a trace-capable `sui` binary. See the [Debugging](/references/ide/debugger) page for more information.

### Syntax highlighting

> Syntax highlighting is provided by the [Move Syntax](https://marketplace.visualstudio.com/items?itemName=damirka.move-syntax) extension.

Reviewer: ```suggestion
The [Move Syntax](https://marketplace.visualstudio.com/items?itemName=damirka.move-syntax) extension provides syntax highlighting.
```

### Hover information

> Hovering over identifiers will show type information and other information about the identifier. This works for all Move symbols including macros. 

Reviewer: ```suggestion
Hovering over identifiers shows type information and other data about the identifier. This works for all Move symbols including macros. 
```

Author: Fixed the tense and made the list more exact.

![Struct hoverover](./images/hover_struct.png)
![Function hoverover](./images/hover_fun.png)
![Macro hoverover](./images/hover_macros.png)

### Code completion

> The move extension will attempt to autocomplete upon a dot operator, displaying the available methods and fields for the type.

Reviewer: ```suggestion
The Move extension attempts to autocomplete upon a dot operator, displaying the available methods and fields for the type.
```

Author: Cleaned it up a little more.

![Code completion](./images/dot_completion.png)

> ### Go-to-definition

Reviewer: ```suggestion
### Definition linking
```

> The move extension supports go-to-definition for all Move symbols including types, functions, and macros, as long as the type was present when move-analyzer last built the file.

Reviewer: ```suggestion
The move extension supports `go-to-definition` linking for all Move symbols including types, functions, and macros, as long as the type was present when move-analyzer last built the file.
```

## Installing

You can install the Move extension from the Visual Studio Code Marketplace:

- Open VS Code
- Open the Extensions view (<kbd>⇧</kbd> + <kbd>⌘</kbd> + <kbd>X</kbd> on macOS, <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>X</kbd> on Windows/Linux) 
- Search for `mysten.move`
- Click **Install** on the Move extension by Mysten Labs

You can also use <kbd>Ctrl</kbd> + <kbd>P</kbd> or <kbd>⌘</kbd> + <kbd>P</kbd> and type `ext install mysten.move` to install the extension, or install via the command line: 

```bash
code --install-extension mysten.move
```

Installing the Move extension also installs the following extensions:

- [Move Syntax](https://marketplace.visualstudio.com/items?itemName=damirka.move-syntax)
- [Move Trace Debugger](https://marketplace.visualstudio.com/items?itemName=mysten.move-trace-debug)

### Installing `move-analyzer`

The Move extension attempts to install the appropriate `move-analyzer` binary for your platform. If this doesn't work, or you prefer to install it manually, build it with Cargo:

```bash
cargo install --git https://github.com/MystenLabs/sui.git sui-move-lsp
```

> By default, the Move extension expects to find the `move-analyzer` binary in `~/.sui/bin`. You can either copy the binary here, or configure the extension to use a different path. 

Reviewer 2: Do we need to worry about specifying the location for windows users? I'm guessing not since the only way they can install it is with choco I'm guessing?

Author: This is also the path on Windows, just interpolate `~` with %HOMEDIR% or whatever it is. By default this is at `C:\Users\%USERNAME%\.sui\bin`, but that could change based on your home drive.
